src/main/java/net/imagej/ops/filter/vesselness/DefaultFrangi.java

```diff
@@ -0,0 +1,282 @@
+import net.imagej.ops.special.computer.AbstractUnaryComputerOp;
+import net.imglib2.Cursor;
+import net.imglib2.RandomAccess;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.outofbounds.OutOfBoundsMirrorFactory;
+import net.imglib2.outofbounds.OutOfBoundsMirrorFactory.Boundary;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.view.Views;
+
+/**
+ * Input is a 2- or 3-dimensional grayscales image.
+ * 
+ * Applies the Frangi Vesselness filter to an image to highlight vessel-like
+ * structures. The spacing parameter refers to the physical distance between
+ * data points in the RandomAccessibleInterval, and can vary for each dimension
+ * of the image.
+ *
+ * @author Gabe Selzer
+ */
+@Plugin(type = Ops.Filter.Vesselness.class)
+public class DefaultFrangi<T extends RealType<T>, U extends RealType<U>>
+		extends AbstractUnaryComputerOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<U>>
+		implements Ops.Filter.Vesselness {
+
+	public final static int MIN_DIMS = 2;
+
+	public final static int MAX_DIMS = 3;
+
+	@Parameter
+	private double[] spacing;
+
+	@Parameter
+	private int scale;
+
+	protected double alpha = 0.5;
+	protected double beta = 0.5;
+
+	protected double minimumVesselness = Double.MIN_VALUE;
+	protected double maximumVesselness = Double.MAX_VALUE;
+
+	public double getMinimumVesselness() {
+		return minimumVesselness;
+	}
+
+	public double getMaximumVesselness() {
+		return maximumVesselness;
+	}
+
+	private double getDistance(RandomAccess<T> ahead, RandomAccess<T> behind, int d) {
+		double distance = 0;
+
+		for (int i = 0; i < d; i++) {
+			double separation = (ahead.getLongPosition(i) - behind.getLongPosition(i)) * spacing[i];
+			if (separation != 0) {
+				distance += (separation * separation);
+			}
+
+		}
+
+		return Math.sqrt(distance);
+	}
+
+	private double derivative(double val1, double val2, double distance) {
+		return ((val2 - val1) / distance);
+	}
+
+	public long numberOfPoints(RandomAccessibleInterval<T> image) {
+		long totalPoints = 1;
+		long[] dimensions = null;
+		image.dimensions(dimensions);
+		for (long length : dimensions) {
+			totalPoints *= length;
+		}
+		return totalPoints;
+	}
+
+	@Override
+	public void compute(final RandomAccessibleInterval<T> input, final RandomAccessibleInterval<U> output) {
+
+		// set spacing if the parameter is not passed.
+		if (spacing == null) {
+			spacing = new double[input.numDimensions()];
+			for (int i = 0; i < input.numDimensions(); i++)
+				spacing[i] = 1;
+		}
+		
+		frangi(input, output, scale);
+	}
+
+	private void frangi(RandomAccessibleInterval<T> in, RandomAccessibleInterval<U> out, int step) {
+
+		// create denominators used for gaussians later.
+		double ad = 2 * alpha * alpha;
+		double bd = 2 * beta * beta;
+
+		// OutOfBoundsMirrorStrategy for use when the cursor reaches the edges.
+		OutOfBoundsMirrorFactory<T, RandomAccessibleInterval<T>> osmf = new OutOfBoundsMirrorFactory<T, RandomAccessibleInterval<T>>(
```

Comment:
You could make this a `@Parameter` of the op and set the `OutOfBoundsMirrorFactory` as default.

Reply:
What would be the benefit of this?

Reply:
Sorry for the delay, @gselzer!

We have made the `OutOfBoundsFactory` a `@Parameter` in other implementations, so you can feed in another one if you like. I understand that it is an additional parameter to provide, that in practice most user will not overwrite. So I am fine with leaving it as it is, for the sake of usability. However, I wanted to state that this is one of the pitfalls of ImageJ1: making implicit
 assumptions about the out-of-bounds strategy without being able to change it...

Maybe @ctrueden has an opinion on that topic as well?